src/jdk.jshell/share/classes/jdk/jshell/SnippetMaps.java

```diff
@@ -164,18 +164,17 @@ String fullClassNameAndPackageToClass(String full, String pkg) {
         List<String> klasses = importSnippets()
                                .filter(isi -> !isi.isStar)
                                .map(isi -> isi.fullname)
-                               .collect(toList());
+                               .toList();
         for (String k : klasses) {
             if (k.equals(full)) {
                 return full.substring(full.lastIndexOf(".")+1, full.length());
             }
         }
-        List<String> pkgs = importSnippets()
+        Stream<String> pkgs = importSnippets()
                                .filter(isi -> isi.isStar)
-                               .map(isi -> isi.fullname.substring(0, isi.fullname.lastIndexOf(".")))
-                               .collect(toList());
-        pkgs.add(0, "java.lang");
-        for (String ipkg : pkgs) {
+                               .map(isi -> isi.fullname.substring(0, isi.fullname.lastIndexOf(".")));
+        pkgs = Stream.concat(Stream.of("java.lang"), pkgs);
+        for (String ipkg : pkgs.toList()) {
```

Comment:
The original code is strictly speaking in error, as it's modifying the list returned from `Collectors.toList()` which is not guaranteed to be mutable. (Obviously, this isn't enforced.) Instead of concatenating the element to the head of the stream and using `Stream.toList()`, though, it might be just as easy to convert the original code to use `collect(Collectors.toCollection(ArrayList::new))` and leave the insertion of "java.lang" at the front at before. Might make things clearer as to what's going on.

I'll defer to the jshell team on their preference here.